settings/settings-impl/src/main/java/com/duckduckgo/settings/impl/SettingsWebViewActivity.kt

```diff
@@ -0,0 +1,182 @@
+import kotlinx.coroutines.flow.launchIn
+import kotlinx.coroutines.flow.onEach
+import kotlinx.coroutines.launch
+import org.json.JSONObject
+import javax.inject.Inject
+import javax.inject.Named
+
+@InjectWith(ActivityScope::class)
+@ContributeToActivityStarter(SettingsWebViewScreenWithParams::class)
+class SettingsWebViewActivity : DuckDuckGoActivity() {
+    @Inject
+    lateinit var userAgentProvider: UserAgentProvider
+
+    @Inject
+    lateinit var webViewCompat: WebViewCompatWrapper
+
+    @Inject
+    lateinit var pixel: Pixel
+
+    private val viewModel: SettingsWebViewViewModel by bindViewModel()
+
+    @Inject
+    lateinit var css: CoreContentScopeScripts
+
+    @Inject
+    @Named("ContentScopeScripts")
+    lateinit var contentScopeScripts: JsMessaging
+
+    private val binding: ActivitySettingsWebviewBinding by viewBinding()
+
+    private val toolbar
+        get() = binding.includeToolbar.toolbar
+
+    @SuppressLint("SetJavaScriptEnabled")
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        val params = intent.getActivityParams(SettingsWebViewScreenWithParams::class.java)
+        val url = params?.url
+        title = params?.screenTitle.orEmpty()
+
+        setContentView(binding.root)
+        setupToolbar(toolbar)
+
+        lifecycleScope.launch {
+            setupWebView()
+            setupCollectors()
+            setupBackPressedDispatcher()
+
+            viewModel.onStart(url)
+        }
+    }
+
+    private fun setupBackPressedDispatcher() {
+        onBackPressedDispatcher.addCallback(
+            this,
+            object : OnBackPressedCallback(true) {
+                override fun handleOnBackPressed() {
+                    if (binding.settingsWebView.canGoBack()) {
+                        binding.settingsWebView.goBack()
+                    } else {
+                        exit()
+                    }
+                }
+            },
+        )
+    }
+
+    private fun setupCollectors() {
+        viewModel.commands
+            .flowWithLifecycle(lifecycle, Lifecycle.State.CREATED)
+            .onEach { processCommand(it) }
+            .launchIn(lifecycleScope)
+    }
+
+    private fun processCommand(command: SettingsWebViewViewModel.Command) {
+        when (command) {
+            is SettingsWebViewViewModel.Command.LoadUrl -> binding.settingsWebView.loadUrl(command.url)
+            SettingsWebViewViewModel.Command.Exit -> exit()
+        }
+    }
+
+    private fun exit() {
+        binding.settingsWebView.stopLoading()
+        binding.settingsWebView.removeJavascriptInterface(contentScopeScripts.context)
+        binding.root.removeView(binding.settingsWebView)
+        binding.settingsWebView.destroy()
+
+        finish()
+    }
+
+    @SuppressLint("SetJavaScriptEnabled")
+    private suspend fun setupWebView() {
+        binding.settingsWebView.let {
+            contentScopeScripts.register(
```

Comment:
I would either comment out the register and addDocumentStartJavaScript methods since we don't really need to do this yet based on the outcome of the mid-mortem or probably even better I'd feature flag it so we can enable/disable at will and test if this works out or not. (this would be that second feature flag, would call it something like `settingsWebViewCompatApi`